src/sentry/region_to_control/producer.py

```diff
@@ -1,43 +1,88 @@
+import atexit
 import dataclasses
-from typing import Optional
+from typing import Any, Optional
 
+from arroyo import Topic
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer
 from django.conf import settings
 
-from sentry.models import UserIP
+from sentry.models import AuditLogEntry, UserIP
 from sentry.region_to_control.messages import RegionToControlMessage, UserIpEvent
 from sentry.silo import SiloMode
 from sentry.utils import json, kafka_config
-from sentry.utils.pubsub import KafkaPublisher
 
 
 def produce_user_ip(event: UserIpEvent):
     if _should_produce_to_kafka():
-        get_user_ip_publisher().publish(
-            settings.KAFKA_REGION_TO_CONTROL,
-            json.dumps(dataclasses.asdict(RegionToControlMessage(user_ip_event=event))),
+        write_region_to_control_message(
+            dataclasses.asdict(RegionToControlMessage(user_ip_event=event)), sync=False
         )
     else:
         UserIP.objects.create_or_update(
             user_id=event.user_id, ip_address=event.ip_address, values=dataclasses.asdict(event)
         )
 
 
-_user_ip_publisher: Optional[KafkaPublisher] = None
+def produce_audit_log_entry(entry: AuditLogEntry):
+    if _should_produce_to_kafka():
+        write_region_to_control_message(
+            dataclasses.asdict(RegionToControlMessage(audit_log_event=entry.as_kafka_event())),
+            sync=True,
+        )
+    else:
+        entry.save()
+
+
+_publisher: Optional[KafkaProducer] = None
+_empty_headers = list()
+
+
+# TODO: Would be nice to configure the timeout via a dynamic setting.
+def write_region_to_control_message(payload: Any, sync=True, timeout=0.1):
```

Comment:
@fpacifici So this is the main change, fairly small.  This code path is triggered exclusively by the system_audit_log and user_audit_log functions in Region mode (not saas), and those code paths are, as I mentioned, on the inside of transactions that usually involve either subscription changes, spike protection, or org changes.  I'm pondering the timeout.  Ideally this seems like something that would be run time configurable, as well. 

This utility method ends up being used both by the audit logs (for sync writes that need to be durable) as well as

Reply:
the async form which is used by user ips (not as much an issue when those are dropped; they merely represent best effort tracking of user usage.)